clouddriver-openstack/src/main/groovy/com/netflix/spinnaker/clouddriver/openstack/client/OpenstackLoadBalancerV2Provider.groovy

```diff
@@ -0,0 +1,208 @@
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ *  See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package com.netflix.spinnaker.clouddriver.openstack.client
+
+import com.netflix.spinnaker.clouddriver.openstack.deploy.exception.OpenstackProviderException
+import com.netflix.spinnaker.clouddriver.openstack.domain.HealthMonitor
+import org.openstack4j.api.Builders
+import org.openstack4j.model.common.ActionResponse
+import org.openstack4j.model.network.NetFloatingIP
+import org.openstack4j.model.network.Port
+import org.openstack4j.model.network.builder.PortBuilder
+import org.openstack4j.model.network.ext.HealthMonitorType
+import org.openstack4j.model.network.ext.HealthMonitorV2
+import org.openstack4j.model.network.ext.LbMethod
+import org.openstack4j.model.network.ext.LbPoolV2
+import org.openstack4j.model.network.ext.ListenerProtocol
+import org.openstack4j.model.network.ext.ListenerV2
+import org.openstack4j.model.network.ext.LoadBalancerV2
+import org.openstack4j.model.network.ext.Protocol
+
+class OpenstackLoadBalancerV2Provider implements OpenstackLoadBalancerProvider, OpenstackRequestHandler, OpenstackIdentityAware {
+  OpenstackIdentityProvider identityProvider
+
+  OpenstackLoadBalancerV2Provider(OpenstackIdentityProvider identityProvider) {
+    this.identityProvider = identityProvider
+  }
+
+  @Override
+  LoadBalancerV2 createLoadBalancer(final String region, final String name, final String description, final String subnetId) {
+    handleRequest {
+      getRegionClient(region).networking().lbaasV2().loadbalancer().create(Builders.loadbalancerV2()
+        .name(name)
+        .description(description)
+        .subnetId(subnetId)
+        .build())
+    }
+  }
+
+  @Override
+  LoadBalancerV2 getLoadBalancer(final String region, final String id) {
+    LoadBalancerV2 result = handleRequest {
+      getRegionClient(region).networking().lbaasV2().loadbalancer().get(id)
+    }
+
+    if (!result) {
+      throw new OpenstackProviderException("Unable to find load balancer ${id} in ${region}")
+    }
+    result
+  }
+
+  @Override
+  ListenerV2 createListener(final String region, final String name, final String externalProtocol, final Integer externalPort, final String description, final String loadBalancerId) {
+    handleRequest {
+      getRegionClient(region).networking().lbaasV2().listener().create(Builders.listenerV2()
+        .name(name)
+        .description(description)
+        .loadBalancerId(loadBalancerId)
+        .protocolPort(externalPort)
+        .protocol(ListenerProtocol.forValue(externalProtocol))
+        .adminStateUp(Boolean.TRUE)
+        .build())
+    }
+  }
+
+  @Override
+  ListenerV2 getListener(final String region, final String id) {
+    handleRequest {
+      getRegionClient(region).networking().lbaasV2().listener().get(id)
```

Comment:
Yep ... I am really starting to despise this design choice by having to wrap all calls in try/catch blocks.